caluma/extensions/visibilities.py

```diff
@@ -1,8 +1,24 @@
 from caluma.caluma_core.types import Node
 from caluma.caluma_core.visibilities import BaseVisibility, filter_queryset_for
+from caluma.caluma_form.schema import Document, Form, Option, Question
+from caluma.caluma_workflow.schema import Case, Flow, Task, Workflow, WorkItem
 
 
 class MySAGWVisibility(BaseVisibility):
     @filter_queryset_for(Node)
-    def filter_queryset_for_all(self, node, queryset, info):
-        return queryset
+    def filter_queryset(self, node, queryset, info):
+        user = info.context.user
+
+        if "sagw" in user.groups or "admin" in user.groups:
+            return queryset
+
+        if node in [Task, Flow, Workflow, Form, Option, Question]:
+            return queryset
+
+        if node in [WorkItem]:
+            return queryset.filter(assigned_users__contains=[user.username])
```

Comment:
What's the workflow pattern? If user A creates a work item and assigns it to user B, will they still need to see it?

Reply:
As discussed in person.